pkg/services/folder/folderimpl/folder.go

```diff
@@ -460,14 +497,19 @@ func (s *Service) getRootFolders(ctx context.Context, q *folder.GetChildrenQuery
 }
 
 // GetSharedWithMe returns folders available to user, which cannot be accessed from the root folders
-func (s *Service) GetSharedWithMe(ctx context.Context, q *folder.GetChildrenQuery) ([]*folder.Folder, error) {
+func (s *Service) GetSharedWithMe(ctx context.Context, q *folder.GetChildrenQuery, forceLegacy bool) ([]*folder.Folder, error) {
 	start := time.Now()
-	availableNonRootFolders, err := s.getAvailableNonRootFolders(ctx, q)
+	availableNonRootFolders, err := s.getAvailableNonRootFolders(ctx, q, forceLegacy)
 	if err != nil {
 		s.metrics.sharedWithMeFetchFoldersRequestsDuration.WithLabelValues("failure").Observe(time.Since(start).Seconds())
 		return nil, folder.ErrInternal.Errorf("failed to fetch subfolders to which the user has explicit access: %w", err)
 	}
-	rootFolders, err := s.GetChildren(ctx, &folder.GetChildrenQuery{UID: "", OrgID: q.OrgID, SignedInUser: q.SignedInUser, Permission: q.Permission})
+	var rootFolders []*folder.Folder
+	if forceLegacy {
```

Comment:
What's the reasoning for parameter here, rather than the feature toggle?

Reply:
The reason is that `GetSharedWithMe` will be called both by `Get` and `GetLegacy`.

If the feature toggle is enabled and `GetSharedWithMe` get's executed by `GetLegacy` then it needs to query the `sqlstore` and not unified storage. The only way to know that is by having this parameter set by the caller (`Get` and `GetLegacy`)

The alternative is to create another version of `GetSharedWithMe` that points to the unified storage, called `GetSharedWithMeFromApiserver`

What do you think?

Reply:
Makes sense to keep as is! Thanks!